src/AutoRest.CSharp/Common/Generation/Writers/ModelFactoryWriter.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using AutoRest.CSharp.Common.Output.Builders;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Output.Models.Types;
+
+namespace AutoRest.CSharp.Generation.Writers
+{
+    internal static class ModelFactoryWriter
+    {
+        public static IReadOnlyCollection<SchemaObjectType> ObjectTypesThatRequireFactory(IEnumerable<TypeProvider> models)
```

Comment:
Can we put this logic onto the SchemaObjectType itself?

Reply:
Just this method or the whole `ModelFactoryWriter`?

Reply:
The logic around knowing whether the type needs a model factory method.

Reply:
Fixed